templates/custom-resources/alb-rule-priority-generator.js

```diff
@@ -0,0 +1 @@
+'use strict';const aws=require("aws-sdk");let defaultResponseURL,report=function(a,b,c,d,e,f){return new Promise((g,h)=>{const i=require("https"),{URL:j}=require("url");var k=JSON.stringify({Status:c,Reason:f,PhysicalResourceId:d||b.logStreamName,StackId:a.StackId,RequestId:a.RequestId,LogicalResourceId:a.LogicalResourceId,Data:e});const l=new j(a.ResponseURL||defaultResponseURL),m={hostname:l.hostname,port:443,path:l.pathname+l.search,method:"PUT",headers:{"Content-Type":"","Content-Length":k.length}};i.request(m).on("error",h).on("response",a=>{a.resume(),400<=a.statusCode?h(new Error(`Error ${a.statusCode}: ${a.statusMessage}`)):g()}).end(k,"utf8")})};const calculateNextRulePriority=async function(a){var b,c=new aws.ELBv2,d=[];do{const e=await c.describeRules({ListenerArn:a,Marker:b}).promise();d=d.concat(e.Rules),b=e.NextMarker}while(b);let e=1;if(0<d.length){const a=d.map(a=>"default"===a.Priority?0:parseInt(a.Priority)),b=Math.max(...a);e=b+1}return e};exports.nextAvailableRulePriorityHandler=async function(a,b){var c,d,e={};try{switch(a.RequestType){case"Create":d=await calculateNextRulePriority(a.ResourceProperties.ListenerArn),e.Priority=d,c=`alb-rule-priority-${a.LogicalResourceId}`;break;case"Update":case"Delete":c=a.PhysicalResourceId;break;default:throw new Error(`Unsupported request type ${a.RequestType}`);}await report(a,b,"SUCCESS",c,e)}catch(d){console.log(`Caught error ${d}.`),await report(a,b,"FAILED",c,null,d.message)}},exports.withDefaultResponseURL=function(a){defaultResponseURL=a};
```

Comment:
Umm these files should exist if you do any of "make build" or "make test" correctly before committing.

Reply:
> Omg this is such a great commit! Really appreciate this change, it's gonna be so good.
> 
> A couple of small things:
> 
> 1. Can you un-commit the templates/* files? They're generated at build time so we don't need to commit them 😁
> 2. Can you also update the End to End Tests ( e2e/*) and find any code that looks like: `			resp, fetchErr := http.Get(fmt.Sprintf("http://%s/%s/", app.Routes[0].URL, app.Routes[0].Path))` and have it just use the URL?
> 3. Also, just an FYI, we use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) for our git commits so we can automatically generate clean release notes :D  I think PH update this PR, but just a future reference!
> 
> Thank you!

Thank you! Should I put `templates/` into the .gitignore file?

Reply:
No you don't. Those files in `templates/` should not be changed.

Reply:
Ah, got it. Now I see what happened. Thanks!